p2p/src/authenticated/lookup/config.rs

```diff
@@ -30,9 +30,9 @@ pub struct Config<C: Signer> {
     /// is selected from the results (shuffled for each dial attempt).
     pub allow_dns: bool,
 
-    /// Whether or not to attempt handshakes on incoming connections
-    /// from unregistered IP addresses.
-    pub attempt_unregistered_handshakes: bool,
+    /// Whether or not to skip IP verification for incoming connections
+    /// (allows known peers to connect from unknown IPs).
+    pub allow_unknown_ips: bool,
```

Comment:
We could offer the option for peers to register with a `None` egress but that seems like it would encourage the wrong behavior (we should attempt to use a known egress to avoid handshakes with random peers).